website_sale_preset_carts/models/time_frame.py

```diff
@@ -147,6 +147,7 @@ def generate_sale_orders(self):
             customers = (
                 self.env['res.partner']
                     .search([
+                        ('subscription_id', '!=', False),
                         ('subscription_id', '=', frame.subscription_id.id)
                         ])
```

Comment:
this looks weird to me. could you explain why you add ('subscription_id', '!=', False)

Reply:
See #10 !

Reply:
I had to add this because when no `subscription_id` was set, `frame.subscription_id` returned `subscription()` (empty recordset*) and hence `frame.subscription_id.id` returned `False`. This added all partners as customer.

You can test this yourself by placing a breakpoint just before this line during and read it out as it's launched by the newly added test `test_timeframe_nosubscription`.

* I must say I was quite surprised two that this could return an empty recordset and not False. Is this normal?

Reply:
yes it's a normal behavior for Odoo to return empty recordset when no result has been found. I toke the habit to test the result with len(recordset) > 0